CODING_STANDARDS.md

```diff
@@ -58,19 +59,121 @@ Keeping modules to a single responsibility makes the code easier to test, consum
 ES6 modules offer a straightforward way to organize code into logical, granular units. 
 Ideally, individual files are 200 - 300 lines of code.
 
+As a rule of thumb, once a file draws near 400 lines (barring abnormally long constants / comments), 
+start considering how to refactor into smaller pieces. 
+
 #### Less is more
 Once a feature is released, it never goes away. We should avoid adding features that don't offer 
 high user value for price we pay both in maintenance, complexity, and payload size. When in doubt, 
 leave it out. 
 
 This applies especially so to providing two different APIs to accomplish the same thing. Always 
-prefer sticking to a _single_ API for accomplishing something. 
+prefer sticking to a _single_ API for accomplishing something.
+
+### 100 column limit 
+All code and docs in the repo should be 100 columns or fewer. This applies to TypeScript, SCSS,
+ HTML, bash scripts, and markdown files.
 
 ### TypeScript
 
-#### Provide function descriptions
-For functions that are more complicated than a simple getter/setter, provide at least a brief 
-sentence explaining what the function does and/or _why_ it does something.
+#### Typing
+Avoid `any` where possible. If you find yourself using `any`, consider whether a generic may be
+appropriate in your case.
+
+For methods and properties that are part of a component's public API, all types must be explicitly
+specified because our documentation tooling cannot current infer types in places where TypeScript 
+can.
+
+#### Fluent APIs
+When creating a fluent or builder-pattern style API, use the `this` return type for methods:
+```
+class ConfigBuilder {
+  withName(name: string): this {
+    this.config.name = name;
+    return this;
+  }
+}
+```
+
+#### Access modifiers
+* Omit the `public` keyword as it is the default behavior.
+* Use `private` when appropriate and possible, prefixing the name with an underscore.
+* Prefix *library-internal* properties and methods with an underscore without using the `private` 
+keyword. This is necessary for anything that must be public (to be used by Angular), but should not
+be part of the user-facing API. This typically applies to symbols used in template expressions, 
+`@ViewChildren` / `@ContentChildren` properties, host bindings, and `@Input` / `@Output` properties
+(when using an alias).
+
+Additionally, the `@docs-private` JsDoc annotation can be used to hide any symbol from the public
+API docs.
+
```

Comment:
What about the `protected` keyword? Do we prefer to make it public instead? A small note about best practices may be beneficial.